spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/CredentialsProperty.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core;
+
+import java.util.List;
+
+import org.springframework.core.io.Resource;
+
+/**
+ * To be extended by property class subclasses and allow e.g., a "credentials.location" notation.
+ *
+ * @author João Andre Martins
+ */
+public abstract class CredentialsProperty {
```

Comment:
`AbstractCredentials` maybe for the name?

Reply:
Well, my idea was about something like:
```
@ConfigurationProperties("spring.cloud.gcp")
public class GcpProperties extends BaseGcpProperties {
```
Where that `BaseGcpProperties` will contain inner public `Credentials`.

WDYT?
Doesn't it work?

Reply:
Renamed to `AbstractCredentialsProperty`. Kept `Property` in the name just so it isn't confused with a usable `Credentials` object that can be passed to an API.